In [10]:
import pandas as pd
import numpy as np 

Importing data

In [2]:
df = pd.read_csv("/Users/Cos-Ibe/GitHub/Summer 2022 Data Science Intern Challenge /2019 Winter Data Science Intern Challenge Data Set - Sheet1.csv")

In [3]:
df

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0            1       53      746           224            2           cash   
1            2       92      925            90            1           cash   
2            3       44      861           144            1           cash   
3            4       18      935           156            1    credit_card   
4            5       18      883           156            1    credit_card   
...        ...      ...      ...           ...          ...            ...   
4995      4996       73      993           330            2          debit   
4996      4997       48      789           234            2           cash   
4997      4998       56      867           351            3           cash   
4998      4999       60      825           354            2    credit_card   
4999      5000       44      734           288            2          debit   

               created_at  
0     2017-03-13 12:36:56  
1     2017-03-03 17:38:52  
2      2017-03-14 4:23:56  
3     2017-03-26 12:43:37  
4      2017-03-01 4:35:11  
...                   ...  
4995  2017-03-30 13:47:17  
4996  2017-03-16 20:36:16  
4997   2017-03-19 5:42:42  
4998  2017-03-16 14:51:18  
4999  2017-03-18 15:48:18  

[5000 rows x 7 columns]

#### Exploring the dataset
#### The dataset contains two measurable numerical columns, order_amount and total_items, and I used the describe function for summary statistics


In [6]:
df['order_amount'].describe()

count      5000.000000
mean       3145.128000
std       41282.539349
min          90.000000
25%         163.000000
50%         284.000000
75%         390.000000
max      704000.000000
Name: order_amount, dtype: float64

In [7]:
df['total_items'].describe()

count    5000.00000
mean        8.78720
std       116.32032
min         1.00000
25%         1.00000
50%         2.00000
75%         3.00000
max      2000.00000
Name: total_items, dtype: float64

<p> Examining the summary anlysis for order_amount and total_item, it appears the error stemmed from calculating the average order value by taking the average of the order_amount column (3145.13).</p>


<p> According to this <a optimizely.com/optimization-glossary/average-order-value/#:~:text=To%20calculate%20your%20company's%20average,September's%20monthly%20AOV%20was%20%2431" target="_blank">optimizely.com</a>, Average order value (AOV) = the total revenue divided by the total number of orders, so Calculating the average order value by finding the average order_amount is wrong.</p> 


<p> I created a dataframe 'aov' to hold the average order value of each 100 stores </p>

In [16]:
aov = df.groupby(['shop_id'])['order_amount', 'total_items'].sum()

In [23]:
aov['average_order_value'] = aov['order_amount'] / aov['total_items']
aov

order_amount  total_items  average_order_value
shop_id                                                
1               13588           86                158.0
2                9588          102                 94.0
3               14652           99                148.0
4               13184          103                128.0
5               13064           92                142.0
...               ...          ...                  ...
96              16830          110                153.0
97              15552           96                162.0
98              14231          107                133.0
99              18330           94                195.0
100              8547           77                111.0

[100 rows x 3 columns]

<p> Using summary staticts on the "average_order_value" column </P>

In [24]:
aov['average_order_value'].describe()

count      100.000000
mean       407.990000
std       2557.462906
min         90.000000
25%        132.750000
50%        153.000000
75%        168.250000
max      25725.000000
Name: average_order_value, dtype: float64

<p> AOV of $407.99 looks realistic but we know shopify sells affordable shoes, so the average order value is still high. <br>
There is also a huge gap between the minimum and the maximum order value.
WE explore further</p>

In [28]:
pd.set_option("max_rows", None)
aov.sort_values(by='average_order_value', ascending=False)

order_amount  total_items  average_order_value
shop_id                                                
78            2263800           88              25725.0
42           11990176        34063                352.0
12              18693           93                201.0
89              23128          118                196.0
99              18330           94                195.0
50              17756           92                193.0
38              13680           72                190.0
51              16643           89                187.0
6               22627          121                187.0
11              17480           95                184.0
79              17738           98                181.0
43              19367          107                181.0
90              19758          111                178.0
59              21538          121                178.0
60              16461           93                177.0
82              14691           83                177.0
81              22656          128                177.0
17              17600          100                176.0
88              17776          101                176.0
26              16720           95                176.0
70              20241          117                173.0
33              15051           87                173.0
85              11524           67                172.0
55              15732           92                171.0
27              18083          107                169.0
95              12432           74                168.0
46              14940           90                166.0
73              19470          118                165.0
28              13776           84                164.0
71              21320          130                164.0
35              17056          104                164.0
29              19234          118                163.0
19              20538          126                163.0
97              15552           96                162.0
66              16583          103                161.0
40              14168           88                161.0
62              13280           83                160.0
72              14240           89                160.0
91              17600          110                160.0
13              21760          136                160.0
61              17222          109                158.0
1               13588           86                158.0
16              11076           71                156.0
18              17472          112                156.0
77              14040           90                156.0
23              17472          112                156.0
76              13485           87                155.0
65              17864          116                154.0
30              16524          108                153.0
84              20196          132                153.0
15              16065          105                153.0
74              11628           76                153.0
96              16830          110                153.0
87              15198          102                149.0
10              17612          119                148.0
3               14652           99                148.0
57              15729          107                147.0
22              13140           90                146.0
52              12994           89                146.0
47              12180           84                145.0
80              13485           93                145.0
44              10224           71                144.0
37              16330          115                142.0
45              15620          110                142.0
5               13064           92                142.0
21              14200          100                142.0
24              17640          126                140.0
58              15042          109                138.0
63              15368          113                136.0
68              11968           8

<p> Sorting the dataframe by average_order_value column in desecinding order reveals that shop id 78 only sold 88 pairs of sneakers to make 2263800 dollars. <br> <br>
Since each of these shops sells only one model of shoe, we know average order value = cost of shoe. <br> 
store id 78's selling sneakers that cost 25725.0 is quite unusual, it is an outlier. <br> <br>
We can use the mode and the medain function to reduce the effect of store 78 on the analysis </p>


In [33]:
# checking for median
aov['average_order_value'].median()

153.0

In [34]:
# checking for mode
aov['average_order_value'].mode()

0    153.0
dtype: float64

<p> The median of the average order value of the 100 shops is 153 and the mode of the the average order value of the 100 shops is also 153 <br> <br>
I would report the mode of the average order value it reduces the effect of outliers. <br> <br>
Its value is $153. </p>
